packages/aws-cdk-lib/aws-ec2/lib/cfn-init.ts

```diff
@@ -311,10 +311,14 @@ function deepMerge(target?: Record<string, any>, src?: Record<string, any>) {
       if (target[key] && !Array.isArray(target[key])) {
         throw new Error(`Trying to merge array [${value}] into a non-array '${target[key]}'`);
       }
-      target[key] = Array.from(new Set([
-        ...target[key] ?? [],
-        ...value,
-      ]));
+      if (key != 'command') { // don't deduplicate command arguments
+        target[key] = Array.from(new Set([
+          ...target[key] ?? [],
+          ...value,
+        ]));
+      } else {
+        target[key] = value;
+      }
```

Comment:
I'm not entirely sure, but it may be worth updating so nothing slips through the cracks. I can change the logic on line 320 to be similar to line 315, though instead of creating the array from a new `Set`, I can use a new `Array`.